src/ott/initializers/linear/initializers.py

```diff
@@ -313,3 +332,108 @@ def body_fn(i: int, f: jnp.ndarray) -> jnp.ndarray:
     return f.at[i].set(new_f)
 
   return jax.lax.fori_loop(0, len(f), body_fn, f)
+
+
+@functools.partial(jax.jit, static_argnums=(3,))
+def subsample(
+    rng: jax.random.PRNGKey, array: jnp.ndarray, weight: jnp.ndarray,
+    subsample_n: int
+):
+  """Subsample point cloud to uniform weighted points.
+
+  Args:
+    rng: random number generator.
+    array: point cloud, array of size N x d.
+    weight: array of size N.
+    subsample_n: number of points to subsample.
+
+  Returns:
+    subsampled point cloud, array of size subsample_n x d.
+  """
+  N = array.shape[0]
+  indices = jax.random.choice(
+      key=rng, a=N, shape=(subsample_n,), replace=True, p=weight, axis=0
+  )
+  return array[indices]
+
+
+@jax.tree_util.register_pytree_node_class
+class SubsampleInitializer(DefaultInitializer):
```

Comment:
This should inherit from `SinkhornInitializer`, not `DefaultInitializer`.

Reply:
I think DefaultInitializer is correct as we want to only change init_a method and keep init_b. Similar to other initialisers inheriting from Default

Reply:
lmk if I've missed something before resolving this

Reply:
You're right, it's correct, thanks!